FSHD_CLI/FSHD/FSHD.py

```diff
@@ -0,0 +1,344 @@
+                        help="Specify the population of the patient.")
+    group2 = parser.add_argument_group('optional arguments')
+    group2.add_argument("-H", "--haplotypes",  action="store_true",
+                        help="Use this flag to display a list of all possible haplotypes.")
+    group2.add_argument("-n", "--top", type=int, metavar="N",
+                        help="Use this option to only display the top N results.\nIf not specified, all results are displayed.")
+    group2.add_argument("-A", "--add", type=str, metavar="INPUT FILE",
+                        help="Use this argument independently to add new haplotype files to the database. Example usage: -A ./new_haplotypes.json.")
+    group2.add_argument("-o", "--output", type=str, metavar="OUTPUT FILE",
+                        help="Specify the path to an output file where the results will be written.\nIf not specified, results will be printed to the console.")
+    group2.add_argument("-sep", "--separator", type=str, choices=[",", "t", "|", ";", " "], default=",", metavar="SEPARATOR",
+                        help="Choose a separator for the output file.\nOptions include comma (,), tab (t), pipe (|), semicolon (;), or space ( ). Default is comma ','.")
+    group2.add_argument("-l", "--list_datasets", action="store_true",
+                        help="Shows a list of all populations which have a haplotype dataset")
+
+    args = parser.parse_args()
+    return args, parser
+
+
+def run(args):
+    """Function which calls the predict method after parsing of arguments
+    :param args: _description_
+    """
+    args.selection = sorted(args.selection)
+    separator = {'t': '\t', ',': ',', '|': '|',
+                 ';': ';', ' ': ' '}.get(args.separator)
+    result, totaalkans = predict(args)
+    if result != 1 and totaalkans != 1:
+        tot_perm = [100*i[0]/totaalkans for i in result if i[2] > 0]
+        output = args.output is not None
+        filename = args.output if output else None
+        if args.top:
+            result_table(result, totaalkans, args.top,
+                         separator, output, filename)
+        else:
+            result_table(result, totaalkans, len(
+                result), separator, output, filename)
+        if output:
+            with open(args.output, 'a') as output_file:
+                output_file.write(f"Total Likelihood: {sum(tot_perm):.1f}")
+        else:
+            print(f"Total Likelihood: {sum(tot_perm):.1f}")
+    else:
+        print(f"No results found for this selection {' '.join(str(i) for i in args.selection)} for population {args.population}!\n"
+              f"For more information use the -h flag.")
+
+
+def main():
+    args, parser = parser_args()
+    if args.selection and len(args.selection) != 4:
+        parser.error(
+            f"You must have a input of 4 SSLP's. The length of your input is {len(args.selection)} SSLP's.")
+    if validate_combinations(args):
```

Comment:
The function lets me do:
`python3 FSHD.py -s 1 2 3 4 -p African -A haplotypes.json`
I was looking for the documented method of providing an alternative JSON file, but it was trying to add `haplotypes.json` to itself.
Valid combinations are a nice way to check, but you also need to exclude certain combinations.